langchain-core/src/utils/signal.ts

```diff
@@ -0,0 +1,17 @@
+export async function raceWithSignal<T>(
+  promise: Promise<T>,
+  signal?: AbortSignal
+): Promise<T> {
+  if (signal === undefined) {
+    return promise;
+  }
+  if (signal.aborted) {
```

Comment:
@nfcampos If we have this check here, do we need all the `signal.throwIfAborted()` calls elsewhere?